hudi-spark/src/main/scala/org/apache/hudi/DataSourceOptions.scala

```diff
@@ -205,6 +205,12 @@ object DataSourceWriteOptions {
   val PRECOMBINE_FIELD_OPT_KEY = "hoodie.datasource.write.precombine.field"
   val DEFAULT_PRECOMBINE_FIELD_OPT_VAL = "ts"
 
+  /**
+   * Field when set to true, will honor ordering while merging two records. If not, incoming record will overwrite
+   * record in storage.
+   */
+  val HONOR_ORDERING_WHILE_MERGING_OPT_KEY = "hoodie.datasource.honor.ordering.while.merging"
+  val DEFAULT_HONOR_ORDERING_WHILE_MERGING_OPT_VAL = "false"
```

Comment:
keep this `true` is more rational for me. From a new user perspective, I don't really understand the difference between `preCombine` and `combineWithLatestValue`. It took me a while to figure this out and originally I thought this is a bug. But we need to think about the impact of the existing user.

Reply:
I really don't like having these payload specific configs, top level in datasource options.